src/os/bluestore/BlueFS.cc

```diff
@@ -2367,7 +2367,17 @@ void BlueFS::_compact_log_dump_metadata_NF(bluefs_transaction_t *t,
       e.bdev = bdev_new;
     }
     dout(20) << __func__ << " op_file_update " << file_ref->fnode << dendl;
-    t->op_file_update(file_ref->fnode);
+
+    // if inflight_flush_size != 0, we need to exclude these unpersistent data from
+    // fnode's size to avoid data inconsistency.
+    if (file_ref->inflight_flush_size > 0) {
```

Comment:
you can simplify this handling and put it into op_file_update:

op_file_update(fnode, size_adj) {
  fnode.size -=adj;
  original op_file_update stuff
  fnode.size += adj;  
}

Reply:
but this may change the interface of op_file_update(). Do you think it's better than this way? Since this size adjust check is only necessary on metadata dump process. For regular fsync() path's metadata update, it is not required.